File: app/src/main/java/com/frogobox/basegameboard2048/ui/StatsActivity.java
package com.frogobox.basegameboard2048.ui;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuItem;

import androidx.viewpager.widget.ViewPager;

import com.frogobox.basegameboard2048.R;
import com.frogobox.basegameboard2048.base.BaseActivity;
import com.frogobox.basegameboard2048.util.helper.ConstHelper;
import com.frogobox.basegameboard2048.view.StatisticsPagerAdapter;
import com.google.android.material.tabs.TabLayout;

import java.io.File;


public class StatsActivity extends BaseActivity {

    String[] TABNAMES = {"4 x 4", "5 x 5", "6 x 6", "7 x 7"};
    private int[] layouts = new int[]{
            R.layout.fragment_statistics,
            R.layout.fragment_statistics,
            R.layout.fragment_statistics,
            R.layout.fragment_statistics
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_statistics);

        setupDetailActivity("");
        getSupportActionBar().setElevation(0f);

        LayoutInflater layoutInflater = (LayoutInflater) getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        StatisticsPagerAdapter mSectionsPagerAdapter = new StatisticsPagerAdapter(layoutInflater, layouts, TABNAMES, this);

        // Set up the ViewPager with the sections adapter.
        ViewPager mViewPager = findViewById(R.id.viewpager);
        mViewPager.setAdapter(mSectionsPagerAdapter);

        TabLayout tabLayout = findViewById(R.id.tablayout);
        tabLayout.setupWithViewPager(mViewPager);
        //tabLayout.setTabTextColors(Color.WHITE,Color.YELLOW);
        setupShowAdsBanner(findViewById(R.id.ads_banner));

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_toolbar_stats, menu);
        //getMenuInflater().inflate(R.menu.menu_stats, menu);
        return true;
        //return false;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.

        //noinspection SimplifiableIfStatement
        switch (item.getItemId()) {
            case R.id.action_reset:
                //    SaveLoadStatistics.resetStats(this);
                //    mSectionsPagerAdapter.refresh(this);

                resetGameStatistics();
                return true;
            case android.R.id.home:
                finish();
                return true;
        }

        return super.onOptionsItemSelected(item);
    }

    public void resetGameStatistics() {
        for (int n = 4; n <= 7; n++) {
            try {
                File file = new File(getFilesDir(), ConstHelper.Const.FILE_STATISTIC + n + ConstHelper.Ext.TXT);
                file.delete();
            } catch (Exception e) {

            }
        }
        finish();
        startActivity(getIntent());
    }

}

File: app/src/main/java/com/frogobox/board/mvvm/stats/StatsActivity.kt
package com.frogobox.board.mvvm.stats

import android.os.Bundle
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuItem
import com.frogobox.board.R
import com.frogobox.board.core.BaseBindingActivity
import com.frogobox.board.databinding.ActivityStatsBinding
import com.frogobox.board.util.SingleConst
import java.io.File

class StatsActivity : BaseBindingActivity<ActivityStatsBinding>() {

    private var TABNAMES = arrayOf("4 x 4", "5 x 5", "6 x 6", "7 x 7")

    private val layouts = intArrayOf(
        R.layout.fragment_stats,
        R.layout.fragment_stats,
        R.layout.fragment_stats,
        R.layout.fragment_stats
    )

    override fun setupViewBinding(): ActivityStatsBinding {
        return ActivityStatsBinding.inflate(layoutInflater)
    }

    override fun setupViewModel() {
    }

    override fun setupOnCreate(savedInstanceState: Bundle?) {
        setupDetailActivity("")
        setupViewPager()
        showAdBanner(binding.ads.adsBanner)
    }

    override fun setupDetailActivity(title: String) {
        super.setupDetailActivity(title)
        supportActionBar!!.elevation = 0f
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_toolbar_stats, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_reset -> {
                resetGameStatistics()
                return true
            }
            android.R.id.home -> {
                finish()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

    private fun setupViewPager() {
        binding.apply {
            val layoutInflater = getSystemService(LAYOUT_INFLATER_SERVICE) as LayoutInflater
            val mSectionsPagerAdapter =
                StatsPager(layoutInflater, layouts, TABNAMES, this@StatsActivity)
            viewpager.adapter = mSectionsPagerAdapter
            tabLayout.setupWithViewPager(viewpager)
        }
    }

    private fun resetGameStatistics() {
        for (n in 4..7) {
            try {
                val file =
                    File(filesDir, SingleConst.Const.FILE_STATISTIC + n + SingleConst.Ext.TXT)
                file.delete()
            } catch (e: Exception) {
            }
        }
        finish()
        startActivity(intent)
    }

}